src/pubsub/ua_pubsub_reader.c

```diff
@@ -2002,8 +2008,10 @@ receiveBufferedNetworkMessage(UA_Server *server, UA_ReaderGroup *readerGroup,
     while(buffer.length > currentPosition) {
         rv = decodeAndProcessNetworkMessageFun(server, readerGroup, connection,
                                                previousPosition, &buffer, &currentPosition);
-        UA_CHECK_STATUS_WARN(rv, return rv, &server->config.logger, UA_LOGCATEGORY_SERVER,
-                             "SubscribeCallback(): receive message failed");
+        if (!((rv == UA_STATUSCODE_BADNOTFOUND) && (previousPosition != currentPosition))) {
```

Comment:
Please reformulate to get rid of the !
rv != UA_STATUSCODE_BADNOTFOUND || previousPosition == currentPosition